app/models/team.rb

```diff
@@ -0,0 +1,4 @@
+class Team < ActiveRecord::Base
+  has_many :users
+  has_many :goals, through: :user
```

Comment:
For this error:
```
team.goals 
#=> ActiveRecord::HasManyThroughAssociationNotFoundError: Could not find the association :user in model Team
#   from /usr/local/rvm/gems/ruby-2.2.3@rails424/gems/activerecord-4.2.4/lib/active_record/reflection.rb:828:in `check_validity!'
```
The interesting part of the error message is: `Could not find the association :user in model Team`

This is because the `through: :user` option (on line 3) is referencing a non-existent association named "user". The association name is "user**s**" (see assocation defined on line 2). Try changing line 3 to:

```
has_many :goals, through: :users
```